src/cli/cmds/init.js

```diff
@@ -0,0 +1,9 @@
+import Init from '../../sub-commands/init';
```

Comment:
Interesting!

So we have a blueprint that covers the general options and dynamically interrogates individual blueprints for their options. And prompts the users for config values for everything.

I've definitely given some thought to interrogating the yargs config for blueprints such that init could dynamically generate the prompts for creating blueprintrc. I don't have a working solution for that yet.

Prompting the user for values for each options is likely something that would need to remain specific to an init implementation rather than having blueprintrc itself be a blueprint that could be generated the same as any other.

Reply:
I'd say prompts them for missing but required values and/or for empty values of options that explicitly require a prompt. 

Is it convenient that we could 
```
.option('use-jsx', {
  alias: 'j',
  describe: "Save javascript files using .jsx",
  type: 'boolean',
  required: true,
  prompt: "JSX extension?"
})
```

as for prompts remaining specific to init.  I disagree.  The only config that we need to generate that blueprint is the path the file is to be written too.  It then prompts for options  that aren't already designed.  You make the init command require an argument or option that determine the path, which is passed to the constructor.

Reply:
Forgive me, I'm not sure I have a clear understanding of your vision.

The idea is that any blueprint can prompt users interactively to provide required options that they haven't provided by means of rc/env/cli? Including one that generates blueprintrc?

To my understanding that is beyond the ability of yargs, although yargs can certainly define options as required and error out if they aren't provided. We can also ensure that values available via rc/env are given to yargs before it makes that decision.

Reply:
Hmmm. Maybe I'm getting a handle on it.

blueprintrc is a blueprint that needs one required parameter for path. I can definitely put an example of that together.

It then prompts for a bunch of stuff, some of which it may be able to read from individual blueprints. Perhaps we could build that into the beforeInstall hook rather than yargs.

Reply:
Probably a point of confusion in that I always meant this to happen at the Sub-Task level and not as part of yargs.

Reply:
So that means required options would be handled by yargs, and prompts for options that are blank and needed by the blueprint.

Reply:
Cool. Thanks. Then I completely agree. I'll look some more at how viable it is to extract the yargs options from each blueprint to feed them into prompts for init. Some initial investigation suggested proved positive.

I may create a branch from my branch and play with that as a separate endeavour.